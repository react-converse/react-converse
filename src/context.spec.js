import React from 'react';
import { mount } from 'enzyme';
import { ConverseContext } from '~/index';

// https://github.com/airbnb/enzyme/issues/1509#issuecomment-401207909
jest.mock('react', () => {
  const originalReact = require.requireActual('react');
  return {
    ...originalReact,
    createContext: require.requireActual('create-react-context').default,
  };
});

describe('ConverseContext', () => {
  it('throws without provider', () => {
    jest.spyOn(console, 'error').mockImplementation(() => {});
    expect(() =>
      mount(
        <ConverseContext.Consumer>
          {converse => converse.startTyping() && <div />}
        </ConverseContext.Consumer>
      )
    ).toThrowErrorMatchingSnapshot();
    console.error.mockReset();
  });
});
