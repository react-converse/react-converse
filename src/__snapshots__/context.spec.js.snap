// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConverseContext throws without provider 1`] = `"No \`Converse\` provider was not found. Make sure to wrap root of your App with it."`;
